src/System.Runtime/tests/System/DateTimeOffsetTests.cs

```diff
@@ -871,6 +871,16 @@ public static void ParseExact_String_String_FormatProvider_DateTimeStyles_O()
         }
 
         [Fact]
+        public static void ParseExact_String_String_FormatProvider_DateTimeStyles_R()
+        {
+            DateTimeOffset expected = DateTimeOffset.MaxValue;
+            string expectedString = expected.ToString("r");
+
+            DateTimeOffset result = DateTimeOffset.ParseExact(expectedString, "r", null, DateTimeStyles.None);
+            Assert.Equal(expectedString, result.ToString("r"));
+        }
```

Comment:
Yeah, I was talking to @tarekgh about this a few days ago. The testing here exercises the code paths but doesn't really give great coverage. What I did here is mimic the code path testing we have for the other formats. I plan to file an issue to track taking a more coherent look at our test coverage. I agree that a theory with a variety of inputs would be best. It should be done for all these tests, not just this format. Min value, Max value, combined with 0, min and max offsets. Now, and some interesting hard coded dates that exercise things in the middle. That should be the data set we feed into all the parsing code.